main/src/ca/uwaterloo/flix/language/phase/jvm/BackendType.scala

```diff
@@ -122,16 +143,69 @@ object BackendType {
   def toErasedBackendType(tpe: MonoType): BackendType = tpe match {
     case MonoType.Bool => Bool
     case MonoType.Char => Char
-    case MonoType.Float32 => Float32
-    case MonoType.Float64 => Float64
     case MonoType.Int8 => Int8
     case MonoType.Int16 => Int16
     case MonoType.Int32 => Int32
     case MonoType.Int64 => Int64
+    case MonoType.Float32 => Float32
+    case MonoType.Float64 => Float64
     case MonoType.Unit | MonoType.BigDecimal | MonoType.BigInt | MonoType.String | MonoType.Regex |
          MonoType.Array(_) | MonoType.Lazy(_) | MonoType.Ref(_) | MonoType.Tuple(_) |
          MonoType.Enum(_) | MonoType.Arrow(_, _) | MonoType.RecordEmpty | MonoType.RecordExtend(_, _, _) |
          MonoType.SchemaEmpty | MonoType.SchemaExtend(_, _, _) | MonoType.Native(_) |
          MonoType.Region => BackendObjType.JavaObject.toTpe
   }
+
+  /**
+    * Computes the `BackendType` based on the given `MonoType`.
+    */
+  def toFlixErasedBackendType(tpe: MonoType): BackendType = tpe match {
+    case MonoType.Bool => Bool
+    case MonoType.Char => Char
+    case MonoType.Int8 => Int8
+    case MonoType.Int16 => Int16
+    case MonoType.Int32 => Int32
+    case MonoType.Int64 => Int64
+    case MonoType.Float32 => Float32
+    case MonoType.Float64 => Float64
+    case MonoType.Array(t) => Array(toFlixErasedBackendType(t))
+    case MonoType.BigDecimal => BackendObjType.BigDecimal.toTpe
+    case MonoType.BigInt => BackendObjType.BigInt.toTpe
+    case MonoType.String => BackendObjType.String.toTpe
+    case MonoType.Regex => BackendObjType.Regex.toTpe
+    case MonoType.Native(clazz) =>
+      // Maybe use clazz.getPackage and clazz.getSimpleName
+      // TODO: Ugly hack.
+      val fqn = clazz.getName.replace('.', '/')
+      BackendObjType.Native(JvmName.mk(fqn)).toTpe
+    case MonoType.Unit | MonoType.Lazy(_) | MonoType.Ref(_) |
+         MonoType.Tuple(_) | MonoType.Arrow(_, _) | MonoType.RecordEmpty |
+         MonoType.RecordExtend(_, _, _) | MonoType.Region | MonoType.Enum(_) |
+         MonoType.SchemaEmpty | MonoType.SchemaExtend(_, _, _) => BackendObjType.JavaObject.toTpe
+  }
+
+  sealed trait PrimitiveType {
```

Comment:
Every `PrimitiveType` is a `BackendType` so let `PrimitiveType` extend `BackendType` and then `Bool` will just extend `PrimitiveType` instead of both. Then you also don't need `toTpe`.